src/routes/mint/[address]/[id].svelte

```diff
@@ -1,15 +1,141 @@
 <script lang="ts">
 	import { page } from '$app/stores';
 	import { onMount } from 'svelte';
-
+	import { env } from '$lib/constants';
+	// import { tokenUriToMetaData } from '$lib/nft-utils';
+	// import { evaluateNft, getTokenUriType, getImageType } from '$lib/utils/functions';
+	// import defaultAbi from '$lib/defaultAbi';
+	// import { Contract, ethers } from 'ethers';
+	import { user } from '$lib/stores/user';
+	
+	let errorPreparingMint = false;
+	let readyToMint = false;
+	let newTokenURI = '';
+	
+	let newMetadata = {};
+	
 	onMount(async () => {
 		// Do initialization things here
+		
+		// const node1 = await Ipfs.create();
+		// const result = await node1.add("hello world2");
+		// debugger;
+		
+		
+		// If you directly load this page, the wallet isn't connected so it can't get blocknumber 
+		let blockNumber = "unknown";
+		if ($user.provider) {
+			blockNumber = await $user.provider.getBlockNumber();
+		}
+		
+		// First, get the token metadata (i.e. result of calling tokenURI and fetching)
+		const metadata = await getTokenMetadata();
+		newMetadata['originalTokenMetadata'] = metadata;
+		newMetadata['description'] = `This is a Snapshot of NFT ${metadata.name} at block #${blockNumber}`;
```

Comment:
Might be a good place to plug our final branding, ie, `This is an NFSnapshot of blah`